src/pages/generate.tsx

```diff
@@ -116,98 +134,362 @@ export default function ClientIdentifierGenerator() {
     });
     setDocumentJson(clientIdDocument);
 
-    const results = await validateLocalDocument(clientIdDocument, localRules);
-    setValidationResults(results);
+    // wait for the component to be re-rendered. Is there a more elegant way?
+    // eslint-disable-next-line no-promise-executor-return
+    await new Promise((r) => setTimeout(r, 200));
+    window.scrollTo({ top: 10000, behavior: "smooth" });
```

Comment:
I think a cleaner way to obtain a similar result is to rely on the `useEffect` hook, with a dependency on the `results` state variable.

Reply:
A great, that's what I was looking for, thanks!